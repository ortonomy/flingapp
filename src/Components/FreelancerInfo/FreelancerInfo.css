.FreelancerInfo {
  margin: 2em;
  background-color: #fafafa;
  height: 100%;
  width: 95%;
}
