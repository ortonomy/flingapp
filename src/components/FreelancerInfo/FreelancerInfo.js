import React, { Component } from 'react';
import InfoTable from '../InfoTable/InfoTable';
import info from './FreelancerInfo.module.css';

const personalInfo = {
  'first name': 'Jorge',
  'last name': 'Brock',
  'location': 'United Kingdom',
  'timezone': 'London, Lisbon ... UTC/GMT + 00:00',
  'native speaker': 'Yes',
  'primary language': 'US English'
}

const proInfo = {
  'capable roles': 'Chief editor, Editor, Content developer',
  'employment status': 'Full time',
  'tags': ['expertise:ESP','works fast'],
  'documents': ['freelancer_assessment.doc']
}

const experience = {}

class FreelancerInfo extends Component {
  render() {
    return (
      <div className={info.FreelancerInfo}>
        <InfoTable title='Personal Information' info={personalInfo}/>
        <InfoTable title='Professional Information' info={proInfo}/>
        <InfoTable title='Experience' info={experience}/>
      </div>
    )
  }
}

export default FreelancerInfo;
