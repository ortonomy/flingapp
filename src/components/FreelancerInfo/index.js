import React, { Component } from 'react';
import InfoTable from '../InfoTable';
import info from './FreelancerInfo.module.css';

class FreelancerInfo extends Component {
  render() {
    const personalInfo = {
      'first name': this.props.freelancer.flFirstName,
      'last name': this.props.freelancer.flLastName,
      'location': this.props.freelancer.flLocation,
      'timezone': this.props.freelancer.flTimezone,
      'native speaker': this.props.freelancer.flIsNativeSpeaker ? 'Yes' : 'No',
      'primary language': this.props.freelancer.flPrimaryLanguage
    }
    const proInfo = {
      'capable roles': 'Chief editor, Editor, Content developer',
      'employment status': this.props.freelancer.flEmploymentStatus,
      'tags': ['expertise:ESP','works fast'],
      'documents': ['freelancer_assessment.doc']
    }
    const experience = {}

    return (
      <div className={info.FreelancerInfo}>
        <InfoTable title='Personal Information' info={personalInfo}/>
        <InfoTable title='Professional Information' info={proInfo}/>
        <InfoTable title='Experience' info={experience}/>
      </div>
    )
  }
}

export default FreelancerInfo;
