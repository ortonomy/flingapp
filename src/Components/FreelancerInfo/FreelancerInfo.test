import React from 'react';
import ReactDOM from 'react-dom';
import FreelancerInfo from './FreelancerInfo';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<FreelancerInfo/>, div);
});
