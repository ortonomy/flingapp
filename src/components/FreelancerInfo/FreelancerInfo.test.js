import React from 'react';
import FreelancerInfo from '../FreelancerInfo';

const freelancer = {
  flFirstName: 'example',
  flLastName: 'freelancer'
}

it('renders without crashing', () => {
  const wrapper = shallow(<FreelancerInfo freelancer={freelancer} />);
  expect(wrapper).toMatchSnapshot();
});
