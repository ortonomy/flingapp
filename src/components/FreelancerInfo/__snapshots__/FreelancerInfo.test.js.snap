// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FreelancerInfo
    freelancer={
        Object {
            "flFirstName": "example",
            "flLastName": "freelancer",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <InfoTable
          info={
                    Object {
                              "first name": "example",
                              "last name": "freelancer",
                              "location": undefined,
                              "native speaker": "No",
                              "primary language": undefined,
                              "timezone": undefined,
                            }
          }
          title="Personal Information"
/>,
        <InfoTable
          info={
                    Object {
                              "capable roles": "Chief editor, Editor, Content developer",
                              "documents": Array [
                                "freelancer_assessment.doc",
                              ],
                              "employment status": undefined,
                              "tags": Array [
                                "expertise:ESP",
                                "works fast",
                              ],
                            }
          }
          title="Professional Information"
/>,
        <InfoTable
          info={Object {}}
          title="Experience"
/>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "info": Object {
            "first name": "example",
            "last name": "freelancer",
            "location": undefined,
            "native speaker": "No",
            "primary language": undefined,
            "timezone": undefined,
          },
          "title": "Personal Information",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "info": Object {
            "capable roles": "Chief editor, Editor, Content developer",
            "documents": Array [
              "freelancer_assessment.doc",
            ],
            "employment status": undefined,
            "tags": Array [
              "expertise:ESP",
              "works fast",
            ],
          },
          "title": "Professional Information",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "info": Object {},
          "title": "Experience",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <InfoTable
            info={
                        Object {
                                    "first name": "example",
                                    "last name": "freelancer",
                                    "location": undefined,
                                    "native speaker": "No",
                                    "primary language": undefined,
                                    "timezone": undefined,
                                  }
            }
            title="Personal Information"
/>,
          <InfoTable
            info={
                        Object {
                                    "capable roles": "Chief editor, Editor, Content developer",
                                    "documents": Array [
                                      "freelancer_assessment.doc",
                                    ],
                                    "employment status": undefined,
                                    "tags": Array [
                                      "expertise:ESP",
                                      "works fast",
                                    ],
                                  }
            }
            title="Professional Information"
/>,
          <InfoTable
            info={Object {}}
            title="Experience"
/>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "info": Object {
              "first name": "example",
              "last name": "freelancer",
              "location": undefined,
              "native speaker": "No",
              "primary language": undefined,
              "timezone": undefined,
            },
            "title": "Personal Information",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "info": Object {
              "capable roles": "Chief editor, Editor, Content developer",
              "documents": Array [
                "freelancer_assessment.doc",
              ],
              "employment status": undefined,
              "tags": Array [
                "expertise:ESP",
                "works fast",
              ],
            },
            "title": "Professional Information",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "info": Object {},
            "title": "Experience",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
